src/client/services/conversation/conversation.js

```diff
@@ -89,6 +92,46 @@ var ConversationService = SparkBase.extend(
     this.encryptionDisabled = false;
   },
 
+  getExifData: function getExifData(file, buf) {
+    return new Promise(function(resolve) {
+      if (file && file.image && file.mimeType === 'image/jpeg') {
+        new ExifImage({ image : buf }, function (error, exifData) {
+          if (error) {
+            this.logger.info('conversation: error with image rotation ', error.message);
+          }
+          else {
+            if (exifData) {
+              file.image.orientation = exifData.image.Orientation;
+            }
+          }
+          resolve(buf);
+        });
+      }
+      else {
+        resolve(buf);
+      }
+    }.bind(this))
+  },
+
+  fixImageOrientation: function fixImageOrientation(file) {
```

Comment:
Yes, this part does the image rotation for the file chips that are not sent in the activity. So there are effectively 3 places where the file orientation needs to be fixed:
1) When the user adds the files but does not send it. The image in the file chip needs to be correctly oriented. This function takes care of that only.
2) When the user clicks on 'Share' button. At that time the image is first correctly rotated and then the thumbnail is created and sent it over to the file share.
3) On image preview, i.e when the user clicks the thumbnail to open the high resolution image in the lightbox. This is handled during the download function of sdk and the orientation details are read and sent along with file data to the web-client, which then correctly rotates it using css.